binderhub/repoproviders.py

```diff
@@ -456,8 +456,8 @@ class GitRepoProvider(RepoProvider):
     <url-escaped-namespace>/<resolved_ref>
 
     eg:
-    https%3A%2F%2Fgithub.com%2Fjupyterhub%2Fzero-to-jupyterhub-k8s/master
-    https%3A%2F%2Fgithub.com%2Fjupyterhub%2Fzero-to-jupyterhub-k8s/f7f3ff6d1bf708bdc12e5f10e18b2a90a4795603
+    https%3A%2F%2Fgithub.com%2Fbinder-examples%2Fconda/main
```

Comment:
I thought it makes more sense to use binder-examples/conda as the example in this doc, but we'll need to rename the default branch on that repo. Since this is the `GitRepoProvider` I think this should use the branch instead of `HEAD` to be applicable to generic Git repos.